appinfo/Migrations/Version20170913113840.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * @author Sujith Haridasan <sharidasan@owncloud.com>
+ *
+ * @copyright Copyright (c) 2017, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Encryption\Migrations;
+
+use OCP\Migration\ISimpleMigration;
+use OCP\Migration\IOutput;
+
+class Version20170913113840 implements ISimpleMigration {
+
+	/**
+	 * @param IOutput $out
+	*/
+	public function run(IOutput $out) {
+		$installedVersion = \OC::$server->getConfig()->getSystemValue('version', '0.0.0');
+
+		if (version_compare('10.0.3', $installedVersion, '>=') === true) {
```

Comment:
Now this is the only modification I made. This is equivalent to `10.0.3 >= $installeddVersion`.

Since we are heading towards 10.0.3, this should be good enough. The UI changed from 10.0.2.
So till version < 10.0.2, the upgrade should work as it did. Or lets say if `userSpecificKey` is added to appconfig for versions less than 10.0.2, then, nothing would prohibit the code flow. 

Now once the upgrade reaches 10.0.2 which is still less than 10.0.3, goes inside the if condition. And then tries to check if the condition is met with the inner `if` condition ( the one with lots of `&&`). And adds the `userSpecificKey` if the condition is true and UI will not have dropdown list.

Lets say if version >= 10.0.2, then the inner if condition may/may not be hit. Because if the encryption is enabled then either of the types would be selected. If not then again falls back to `userSpecificKey`. This logic applies till, including version 10.0.3.

Though technically the version check is redundant imho. Because unless explicitly selected for masterkey or enabled masterkey via command line => its userSpecificKey. Which sticks with the inner if condition.